xbmc/epg/EpgContainer.cpp

```diff
@@ -235,20 +235,16 @@ void CEpgContainer::LoadFromDB(void)
 
 bool CEpgContainer::PersistTables(void)
 {
-  m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
-  m_critSection.unlock();
-  return m_database.Persist(copy);
+  CSingleLock lock(m_critSection);
```

Comment:
I've already said it at least 3 times in this thread that you can't do this. You cannot hold a lock while the Persist() method is being called, because it will block the GUI from accessing the EPG. e.g. if you have the timeline window open while this method is being called when using a slow database, like a raspberry pi 1 with a slow sd card, then the screen will freeze. That's why a copy of the table is persisted here and in the method below, and your patch removed this.